CI: Conformance Gateway API TestConformance/HTTPRouteRequestMultipleMirrors

## CI failure

```
    --- FAIL: TestConformance/HTTPRouteRequestMultipleMirrors (0.07s)
        --- PASS: TestConformance/HTTPRouteRequestMultipleMirrors/0_request_to_'/mirror'_should_go_to_infra-backend-v1 (2.04s)
        --- FAIL: TestConformance/HTTPRouteRequestMultipleMirrors/1_request_to_'/mirror-and-modify-request-headers'_with_headers_should_go_to_infra-backend-v1 (61.02s)
```

zip: [cilium-sysdump-out.zip.zip](https://github.com/cilium/cilium/files/12793771/cilium-sysdump-out.zip.zip)
 
Link: https://github.com/cilium/cilium/actions/runs/6393774281/job/17353698582
Logs:

```
NAME  TestConformance/HTTPRouteRequestMultipleMirrors/1_request_to_'/mirror-and-modify-request-headers'_with_headers_should_go_to_infra-backend-v1
    mirror.go:51: Searching for the mirrored request log
    mirror.go:52: Reading "gateway-conformance-infra/infra-backend-v2" logs
    mirror.go:51: Searching for the mirrored request log
    ...
    [REPEATED MANY TIMES]
    ...
    mirror.go:51: Searching for the mirrored request log
    mirror.go:52: Reading "gateway-conformance-infra/infra-backend-v2" logs
    mirror.go:51: Searching for the mirrored request log
    mirror.go:52: Reading "gateway-conformance-infra/infra-backend-v2" logs
    mirror.go:47: 
        	Error Trace:	/home/runner/work/cilium/cilium/vendor/sigs.k8s.io/gateway-api/conformance/utils/http/mirror.go:47
        	            				/opt/hostedtoolcache/go/1.21.1/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Condition never satisfied
        	Test:       	TestConformance/HTTPRouteRequestMultipleMirrors/1_request_to_'/mirror-and-modify-request-headers'_with_headers_should_go_to_infra-backend-v1
        	Messages:   	Couldn't find mirrored request in "gateway-conformance-infra/infra-backend-v2" logs
    mirror.go:72: Found mirrored request log in all desired backends
```

--- a/.github/workflows/conformance-gateway-api.yaml
+++ b/.github/workflows/conformance-gateway-api.yaml
@@ -118,8 +118,7 @@ jobs:
         run: |
           echo sha=${{ steps.default_vars.outputs.sha }} >> $GITHUB_OUTPUT
           
-          # Enable HTTPRouteRequestMultipleMirrors again once https://github.com/cilium/cilium/issues/28374 is fixed
-          SUPPORTED_FEATURES="ReferenceGrant,HTTPRoute,TLSRoute,HTTPRouteQueryParamMatching,HTTPRouteMethodMatching,GatewayClassObservedGenerationBump,HTTPRouteHostRewrite,HTTPRoutePathRewrite,HTTPRouteSchemeRedirect,HTTPRoutePathRedirect,HTTPRoutePortRedirect,HTTPRouteRequestMirror"
+          SUPPORTED_FEATURES="ReferenceGrant,HTTPRoute,TLSRoute,HTTPRouteQueryParamMatching,HTTPRouteMethodMatching,GatewayClassObservedGenerationBump,HTTPRouteHostRewrite,HTTPRoutePathRewrite,HTTPRouteSchemeRedirect,HTTPRoutePathRedirect,HTTPRoutePortRedirect,HTTPRouteRequestMirror,HTTPRouteRequestMultipleMirrors"
           if [ ${{ matrix.crd-channel }} == "experimental" ]; then
             SUPPORTED_FEATURES+=",HTTPResponseHeaderModification,RouteDestinationPortMatching"
           fi

--- a/operator/pkg/model/translation/envoy_virtual_host.go
+++ b/operator/pkg/model/translation/envoy_virtual_host.go
@@ -12,6 +12,7 @@ import (
 	envoy_config_core_v3 "github.com/cilium/proxy/go/envoy/config/core/v3"
 	envoy_config_route_v3 "github.com/cilium/proxy/go/envoy/config/route/v3"
 	envoy_type_matcher_v3 "github.com/cilium/proxy/go/envoy/type/matcher/v3"
+	envoy_type_v3 "github.com/cilium/proxy/go/envoy/type/v3"
 	"google.golang.org/protobuf/types/known/durationpb"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 
@@ -319,6 +320,11 @@ func requestMirrorMutation(mirrors []*model.HTTPRequestMirror) routeActionMutati
 			}
 			action = append(action, &envoy_config_route_v3.RouteAction_RequestMirrorPolicy{
 				Cluster: fmt.Sprintf("%s/%s:%s", m.Backend.Namespace, m.Backend.Name, m.Backend.Port.GetPort()),
+				RuntimeFraction: &envoy_config_core_v3.RuntimeFractionalPercent{
+					DefaultValue: &envoy_type_v3.FractionalPercent{
+						Numerator: 100,
+					},
+				},
 			})
 		}
 		route.Route.RequestMirrorPolicies = action

--- a/operator/pkg/model/translation/envoy_virtual_host_test.go
+++ b/operator/pkg/model/translation/envoy_virtual_host_test.go
@@ -253,6 +253,8 @@ func Test_requestMirrorMutation(t *testing.T) {
 		res := requestMirrorMutation(mirror)(route)
 		require.Len(t, res.Route.RequestMirrorPolicies, 2)
 		require.Equal(t, res.Route.RequestMirrorPolicies[0].Cluster, "default/dummy-service:8080")
+		require.Equal(t, res.Route.RequestMirrorPolicies[0].RuntimeFraction.DefaultValue.Numerator, uint32(100))
 		require.Equal(t, res.Route.RequestMirrorPolicies[1].Cluster, "default/another-dummy-service:8080")
+		require.Equal(t, res.Route.RequestMirrorPolicies[1].RuntimeFraction.DefaultValue.Numerator, uint32(100))
 	})
 }

--- a/operator/pkg/model/translation/gateway-api/translator_fixture_test.go
+++ b/operator/pkg/model/translation/gateway-api/translator_fixture_test.go
@@ -17,6 +17,7 @@ import (
 	envoy_extensions_filters_network_tcp_v3 "github.com/cilium/proxy/go/envoy/extensions/filters/network/tcp_proxy/v3"
 	envoy_upstreams_http_v3 "github.com/cilium/proxy/go/envoy/extensions/upstreams/http/v3"
 	envoy_type_matcher_v3 "github.com/cilium/proxy/go/envoy/type/matcher/v3"
+	envoy_type_v3 "github.com/cilium/proxy/go/envoy/type/v3"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/anypb"
 	"google.golang.org/protobuf/types/known/durationpb"
@@ -3416,6 +3417,11 @@ var mirrorHTTPListenersCiliumEnvoyConfig = &ciliumv2.CiliumEnvoyConfig{
 											RequestMirrorPolicies: []*envoy_config_route_v3.RouteAction_RequestMirrorPolicy{
 												{
 													Cluster: fmt.Sprintf("%s/%s:%s", "gateway-conformance-infra", "infra-backend-v2", "8080"),
+													RuntimeFraction: &envoy_config_core_v3.RuntimeFractionalPercent{
+														DefaultValue: &envoy_type_v3.FractionalPercent{
+															Numerator: 100,
+														},
+													},
 												},
 											},
 										},

